mwui-stencil/src/utils/delegate.ts

```diff
@@ -0,0 +1,54 @@
+export const attachComponent = async (delegate: any | undefined, container: HTMLElement): Promise<HTMLElement> => {
+  if (delegate) {
+    return delegate.attachViewToDom(container);
+  }
+};
+
+export const detachComponent = (delegate: any | undefined, element: HTMLElement | undefined) => {
+  if (element) {
+    if (delegate) {
+      const container = element.parentElement;
+      return delegate.removeViewFromDom(container, element);
+    }
+    element.remove();
+  }
+  return Promise.resolve();
+};
+
+export const CoreDelegate = () => {
```

Comment:
it's a utility to move elements from where they are in the DOM to the document root/body
basically this could by any type of notification that needs to be displayed on top of everything

Reply:
also i removed the references to modal

Reply:
ah okay, so it's kind of a portal. Nice! this makes my comment about 100vw obsolete ;)